scripts/entity/world/settlements/stronghold_player_base.nut

```diff
@@ -200,103 +219,72 @@ this.stronghold_player_base <- this.inherit("scripts/entity/world/settlement", {
 
 	function getTooltip()
 	{
-		local ret = this.settlement.getTooltip()
-		if (this.isUpgrading()){ 
+		local ret = this.settlement.getTooltip();
+		if (this.isUpgrading())
+		{ 
 			ret.push({
 				id = 99,
 				type = "description",
 				text = format("\n Currently upgrading to a %s", this.getSizeName(true))
 			})
 		}
-		return ret
+		return ret;
 	}
-	
-	function updateShop( _force = false )
-	{
-		//modified vanilla function to allow storage_building to generate free items.
-		if (this.m.LastShopUpdate == 0.0){
-			this.m.LastShopUpdate = this.Time.getVirtualTimeF() - 3 * this.World.getTime().SecondsPerDay
```

Comment:
this `3` would've been a fine extration to settings. `GENERATE_INCOME_EVERY_N_DAYS = 3`